### PR TITLE
Add dump mode to attest-tool.

### PR DESCRIPTION
**Pull request by twitchy-jsonp**

### CODE REVIEW

**Comment by twitchyliquid64** [comment]
CI is failing due to transient issues with GCP + debian mirrors. I'll re-run when it looks like its back.

File: attest/attest-tool/attest-tool.go [hunk 1]
```diff
@@ -122,8 +152,88 @@ func runCommand(tpm *attest.TPM) error {
...
+			break
+		}
+	}
+
+	if pk == nil {
+		return nil, errors.New("no EK available")
+	}
+
+	if err := pem.Encode(&buf, &pem.Block{Type: "RSA PUBLIC KEY", Bytes: x509.MarshalPKCS1PublicKey(pk)}); err != nil {
+		return nil, fmt.Errorf("pem.Encode() failed: %v", err)
```
Comment (ericchiang) [inline attest/attest-tool/attest-tool.go]: nit: seen this a few times, but generally avoid referencing function names in error messages. just describe what failed

```
return nil, fmt.Errorf("pem encoding RSA key failed: %v")
```

File: attest/attest-tool/attest-tool.go [hunk 2]
```diff
@@ -122,8 +152,88 @@ func runCommand(tpm *attest.TPM) error {
...
+}
+
+func rsaEKPEM(tpm *attest.TPM) ([]byte, error) {
+	eks, err := tpm.EKs()
+	if err != nil {
+		return nil, err
+	}
+
+	var (
+		pk  *rsa.PublicKey
```
Comment (ericchiang) [inline attest/attest-tool/attest-tool.go]: Can you just return the raw cert instead?

File: attest/attest-tool/attest-tool.go [hunk 3]
```diff
@@ -122,8 +152,88 @@ func runCommand(tpm *attest.TPM) error {
...
+	k, err := tpm.MintAIK(nil)
+	if err != nil {
+		return nil, fmt.Errorf("MintAIK() failed: %v", err)
+	}
+	defer k.Close(tpm)
+	out.AIK = k.AttestationParameters()
+
+	// Get a quote.
+	if out.Quote.Nonce, err = hex.DecodeString(*nonceHex); err != nil {
+		return nil, err
```
Comment (ericchiang) [inline attest/attest-tool/attest-tool.go]: nit: describe what failed

File: attest/attest-tool/attest-tool.go (hunk 2 above)
Comment (twitchy-jsonp) [reply attest/attest-tool/attest-tool.go]: We can only do that if the device has a cert & that cert is parseable.

**Comment by ericchiang** [comment]
Please squash your commits (remove the "Address review feedback" commit) before merging.

**Pull request merged**